src/Common/src/CoreLib/System/Globalization/DateTimeParse.cs

```diff
@@ -1160,7 +1169,21 @@ private static bool Lex(DS dps, ref __DTString str, ref DateTimeToken dtok, ref
             return true;
         }
 
-        private static bool VerifyValidPunctuation(ref __DTString str)
+        private static Calendar GetJapaneseCalendarDefaultInstance()
```

Comment:
The need to be split out into separate methods to avoid referencing the `JapaneseCalendar` / `TaiwanCalendar` classes from within a method body that cannot be linked out.

Reply:
Worth comments so nobody refactors them back in again someday?

Reply:
> Worth comments so nobody refactors them back in again someday?

Seems like this Linker feature ended up with the same design as dotnet/designs#42 (convert entire methods to a throw or stub them out), except the linking out behavior is defined [far away](https://github.com/mono/linker/pull/596/commits/7f18668e7683e4723a68cc2b1942a46ce553c6d9) from where the code is (and carries the risks Dan is pointing out since we're hardcoding CoreLib implementation details in the linker). The referenced design uses custom attributes because they're self-documenting and harder to miss during refactorings. It also opens this feature up to second/third party developers.

We talked about what we did in .NET Native here: https://github.com/mono/linker/pull/486#issuecomment-471913443. Is there a reason why we're dismissing the custom attribute approach? The FX API committee didn't have objections to the new attribute in principle, but wanted to gather more data on the usefulness. So the attribute currently lives as a private thing here:

https://github.com/dotnet/corefx/blob/4e0c8ec7b124e2a1c5294bee24e61ef42bbd79ec/src/Common/src/System/Runtime/CompilerServices/RemovableFeatureAttribute.cs#L9-L27

The attribute is currently understood by .NET Native only and we have some APIs annotated with it.

We also discussed having a specialized exception type to be thrown when the code was removed: https://github.com/dotnet/corert/blob/697c70b9942f5b5c104f196b399e12dfb8a323bf/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/FeatureRemovedException.cs#L7-L24

Reply:
> Worth comments so nobody refactors them back in again someday?

That should be just a temporary solution, we will be improving linker to detect code involving GlobalizationMode.Invariant automatically, so the method could be renamed and additional attributes are not necessary